### ISSUE TITLE
Label for "Review Profile config" modal is not displayed properly

### ISSUE THREAD

**Issue by zak905** [keycloak/keycloak-ui#3707]
### Area

admin/ui

### Describe the bug

In Keycloak admin, navigate to Authentication -> first broker login -> and click on "settings"  on the review profile step. You will see the folliowing:

![0 0 0 0_8080_admin_master_console_](https://user-images.githubusercontent.com/6265563/199524372-aca099d1-4bac-419a-a729-ed1bb3a25ccb.png)

It seems like AngularJs interpolation is not working (anymore)

### Version

20.0.0

### Expected behavior

The label should display `Update Profile on First Login` instead of  `{{:: 'update-profile-on-first-login' | translate}}`

### Actual behavior

The label is displaying `{{:: 'update-profile-on-first-login' | translate}}`

### How to Reproduce?

In Keycloak admin,  navigate to Authentication -> first broker login -> and click on "settings"  on the "review profile" step (the first step )

**Comment by mposolda** [comment]
@jonkoops I am re-opening . The main codebase PR https://github.com/keycloak/keycloak/pull/15293 is merged and I've created separate issue in the "main codebase" for it https://github.com/keycloak/keycloak/issues/16658, which is closed now. But the UI PR https://github.com/keycloak/keycloak-ui/pull/3870 is not yet merged and hence re-opening this.

**Comment by zak905** [comment]
thanks @mposolda. I have rebased #3870 against the main branch, and also set the PR from draft to ready for review.

**Comment by jonkoops** [comment]
I believe the fix in https://github.com/keycloak/keycloak/pull/15293 was not complete. The config description call for (`idp-create-user-if-unique`) returns the complete and translated strings for it's properties:

![image](https://user-images.githubusercontent.com/695720/215550995-aca839d9-cebd-4e65-83ce-bff66a64a28b.png)


Where as the same call for `idp-review-profile` now returns untranslated strings:

![image](https://user-images.githubusercontent.com/695720/215551163-3bb28eda-3435-4080-99a8-bfd201f84286.png)


Strings that come back from the API that need to be localized should always be localized in the API itself. This is not the responsibility of the UI.

@zak905 I suggest you revise your implementation in https://github.com/keycloak/keycloak/pull/15293 to do the localization on the API endpoint itself.

**Comment by jonkoops** [comment]
Closing this as this is not something we can address in the UI.

**Comment by zak905** [comment]
Hi @jonkoops, 

If returning the localized string/message directly from the API was possible, Keycloak folks would have done it already to avoid this. Few points concerning the approach discussed in https://github.com/keycloak/keycloak/pull/15293:
* An `AuthenticatorFactory` like the one implemented [here](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/authentication/authenticators/broker/IdpReviewProfileAuthenticatorFactory.java) allows you to define dynamic properties which are used to build authenticators config forms dynamically. This is because each Authenticator configuration is different, and it's not really scalable to have a template for each. Therefore they are rendered dynamically based on the properties returned by the `getConfigProperties` method
* The authenticators config forms are created on the client side by some AngularJS code and therefore the localization is only possible from client/UI side
* A keycloak factory in general is initialized only once, and lives through the whole Keycloak application. At the factory creation time, Keycloak may not yet be aware of the user's locale. 
* This is why `{{:: 'update-profile-on-first-login' | translate}}` was used, and hard coded in the property name inside the Java code. 
* The implementation proposed here and in https://github.com/keycloak/keycloak/pull/15293 ensures compatibility between both the legacy console, and the new ui, at the time being. Off course, this fixes only one Authenticator, and more work is needed to do the same for all the them. 

What you are referring to above is not translated/localized text, you can find it here: https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/authentication/authenticators/broker/IdpCreateUserIfUniqueAuthenticatorFactory.java#L104. It is hard coded. 

Closing this PR, leaves https://github.com/keycloak/keycloak/pull/15293 in a broken state. I hope this is enough reasons to convince you to reopen this. Otherwise, the merged PR in Keycloak needs to be reverted, and simple hard coded text can be used as a workaround solution (a solution that was rejected in first place, as you can see from the first comment in the PR). (cc @mposolda @stianst )

**Comment by jonkoops** [comment]
@zak905 sorry for the late reply, this caused a bit of internal discussion. @mposolda created a a topic for this to discuss what we want to do: https://github.com/keycloak/keycloak/discussions/16882

**Comment by zak905** [comment]
interesting discussion, thanks for the update

**Comment by jonkoops** [comment]
I think for now the best approach would be to 'hard code' the English strings for this in the API. At a later point (we need to create an issue for this) the API will be modified so that all messages are properly localized.

**Comment by zak905** [comment]
It make sens. This would do the thing, until the backend based localization is introduced. I see also that the old console has been removed in the past days, so one more good reason to deal with this:) Here is the new PR:  https://github.com/keycloak/keycloak/pull/17062 
Once we have the issue, I can hopefully help with the implementation ( I can't promise but I will do my best)

**Comment by jonkoops** [comment]
Thanks for your patience @zak905, we very much appreciate your contributions.
